src/github.com/matrix-org/dendrite/clientapi/routing/register_test.go

```diff
@@ -132,3 +132,14 @@ func TestFlowCheckingExtraneousIncorrectInput(t *testing.T) {
 		t.Error("Incorrect registration flow verification: ", testFlow, ", from allowed flows: ", allowedFlows, ". Should be false.")
 	}
 }
+
+// Completed flows stages should always be a valid slice header.
+// TestEmptyCompletedFlows checks that sessionsDict returns a slice & not nil.
+func TestEmptyCompletedFlows(t *testing.T) {
+	fakeEmptySessions := newSessionsDict()
+	fakeSessionID := "aRandomSessionIDWhichDoesNotExist"
+
+	if fakeEmptySessions.GetCompletedStages(fakeSessionID) == nil {
```

Comment:
it'd be better to test for `[]`, rather than accepting everything that's not `nil`